Formula/clickhouse.rb

```diff
@@ -0,0 +1,88 @@
+class Clickhouse < Formula
+  desc "Free analytics DBMS for big data with SQL interface"
+  homepage "https://clickhouse.com"
+  url "https://github.com/ClickHouse/ClickHouse/releases/download/v21.11.1.8636-prestable/ClickHouse_sources_with_submodules.tar.gz"
+  sha256 "e95576f1b9f557ae7aa808fbe8959f14fbc86981d809d21e49eb65fe4ebf1142"
+  license "Apache-2.0"
+  head "https://github.com/ClickHouse/ClickHouse.git", branch: "master"
+
+  depends_on "cmake" => :build
+  depends_on "gawk" => :build
+  depends_on "gettext" => :build
+  depends_on "libtool" => :build
+  depends_on "ninja" => :build
+  depends_on "perl" => :build
+  depends_on "python@3.9" => :build
+  depends_on "llvm" => "13"
+
+  def install
+    cmake_args = std_cmake_args.dup
+
+    # It is crucial that CMake config scripts see RelWithDebInfo as a build type,
+    # since the code is only handling it (and Debug) properly.
+    # It is OK if Homebrew infrastructure filters out the debug info-related flags later.
+    cmake_args.reject! { |x| x.start_with?("-DCMAKE_BUILD_TYPE=") }
+    cmake_args << "-DCMAKE_BUILD_TYPE=RelWithDebInfo"
+
+    # Vanilla Clang is the only officially supported compiler.
+    cmake_args << "-DCMAKE_C_COMPILER=#{Formula["llvm"].bin}/clang"
+    cmake_args << "-DCMAKE_CXX_COMPILER=#{Formula["llvm"].bin}/clang++"
+    cmake_args << "-DCMAKE_AR=#{Formula["llvm"].bin}/llvm-ar"
+    cmake_args << "-DCMAKE_RANLIB=#{Formula["llvm"].bin}/llvm-ranlib"
+    cmake_args << "-DOBJCOPY_PATH=#{Formula["llvm"].bin}/llvm-objcopy"
```

Comment:
Does this actually fail to build with system `clang`, `ar`, etc?

Reply:
Yes, system clang (Xcode 13) lags behind vanilla clang 13.